### ISSUE TITLE
react-pose not recognizing keys on first run when using react-router-dom

### ISSUE THREAD

**Issue by cipherbeta** [Popmotion/popmotion#535]
Using React-Pose to animate page transitions inside of a React (create-react-app) project, using react-router-dom to handle routing. When running the react app with `npm start` or `npm run dev`, the first instance of the app will throw an error. When using a similar structure to [the documentation](https://popmotion.io/pose/learn/route-transitions-react-router/), pose throws an error noting that each child needs to have a key. However, if we remove all animation elements, let the server restart, then re-add the animation sections, the Pose will recognize the keys and allow the app to continue working.

As an example instance, the following throws an error that there are no child keys on the first run. Error traces to the PoseGroup below.

```
const Routes = (props) => (
    <Route render={ ({ location }) => (
        <PoseGroup>
            <RouteContainer key={location.key}>
                <Switch location={location}>
                    <Route exact path="/" component={Home} key="home"/>
                    <Route path="/about" component={About} key="about"/>
                    <Route path="/bugs" component={Bugs} key="bugs"/>
                    <Route path="/security" component={Security} key="security"/>
                    <Route path="/aur" component={Aur} key="aur"/>
                    <Route path="/download" component={Download} key="download"/>
                </Switch>
            </RouteContainer>
        </PoseGroup>
    )}/>
)
```

Changing the Routes file to as follows and allowing the app to restart, then changing it back to the first example and refreshing will let Pose recognize the keys.

```
const Routes = (props) => (
    <Switch location={location}>
        <Route exact path="/" component={Home} key="home"/>
        <Route path="/about" component={About} key="about"/>
        <Route path="/bugs" component={Bugs} key="bugs"/>
        <Route path="/security" component={Security} key="security"/>
        <Route path="/aur" component={Aur} key="aur"/>
        <Route path="/download" component={Download} key="download"/>
    </Switch>
)
```

Using react-pose 3.3.6, react 16.5.2, react-router-dom 4.3.1. Specifically if necessary, reproducible with [this repo](https://github.com/cipherbeta/arch-linux-react).

**Comment by Andarist** [comment]
That's because your location has no `key` property and you are trying to use it as `key` for `<RouteContainer/>` here:
https://github.com/cipherbeta/arch-linux-react/blob/507c3c87e3433b433dc18daaf72244313be9a12d/src/routes/index.js#L20

```js
JSON.stringify(location) // "{"pathname":"/","search":"","hash":""}"
```

**Comment by kafein** [comment]
@cipherbeta try location.pathname instead of location.key

**Comment by cipherbeta** [comment]
@kafein That did the trick, thanks!

**Comment by nkov** [comment]
Had the same issue - any idea why the key would be `undefined` sometimes? I guess its more of a question for `react-router` folk but maybe we could update the docs here to use `location.pathname` for the React Router example for clarity.

**Comment by gustafsilva** [comment]
I lost a lot of time with this bug, very strange. :grimacing:
In the first render it seems to me that the location does not receive any keys. :open_mouth:
After the first navigation it starts receiving key values... :confused: 
I used the pathname inves of the key and everything worked out! :+1: :smile:

**Comment by lusid** [comment]
The example on this page should be updated to reflect this:
https://popmotion.io/pose/learn/route-transitions-react-router/

Line 32 in the sandbox editor should be changed to:
`<RouteContainer key={location.pathname}>`

**Comment by gangwarily** [comment]
This seem to be an issue with the newer versions of react-pose. As @lusid pointed out, I noticed that the codesandbox for the react-router tutorial is using version 3.0.2. When I bumped it down to that version, I was no longer getting the HEY LISTEN error.

Hoping that someone on the PopMotion team can point us in the right direction about whether we should be using location.pathname and whether the tutorial needs to be updated.
